packages/lesswrong/lib/collections/posts/helpers.ts

```diff
@@ -155,15 +156,34 @@ export const postGetLastCommentPromotedAt = (post: PostsBase|DbPost):Date|null =
   return post.lastCommentPromotedAt;
 }
 
-export const postCanEdit = (currentUser: UsersCurrent|DbUser|null, post: PostsBase|DbPost): boolean => {
-  return userOwns(currentUser, post) || userCanDo(currentUser, 'posts.edit.all')
+/**
+ * Whether or not the given user is an organizer for the post's group
+ * @param user
+ * @param post
+ * @returns {Promise} Promise object resolves to true if the post has a group and the user is an organizer for that group
+ */
+export const userIsPostGroupOrganizer = async (user: UsersMinimumInfo|DbUser|null, post: PostsBase|DbPost): Promise<boolean> => {
+  const groupId = ('group' in post) ? post.group?._id : post.groupId;
+  if (!user || !groupId)
+    return false
+    
+  const group = await Localgroups.findOne({_id: groupId});
+  return !!group && group.organizerIds.some(id => id === user._id);
 }
 
-export const postCanDelete = (currentUser: UsersCurrent|DbUser|null, post: PostsBase|DbPost): boolean => {
+export const postCanEdit = (currentUser: UsersCurrent|null, post: PostsBase): boolean => {
+  const organizerIds = post.group?.organizerIds;
+  const isPostGroupOrganizer = organizerIds ? organizerIds.some(id => id === currentUser?._id) : false;
+  return userOwns(currentUser, post) || userCanDo(currentUser, 'posts.edit.all') || isPostGroupOrganizer;
+}
+
+export const postCanDelete = (currentUser: UsersCurrent|null, post: PostsBase): boolean => {
   if (userCanDo(currentUser, "posts.remove.all")) {
     return true
   }
-  return userOwns(currentUser, post) && post.draft
+  const organizerIds = post.group?.organizerIds;
+  const isPostGroupOrganizer = organizerIds ? organizerIds.some(id => id === currentUser?._id) : false;
+  return (userOwns(currentUser, post) || isPostGroupOrganizer) && post.draft
 }
```

Comment:
These two are now only used by the frontend - not sure if I should move them elsewhere, since it feels like they don't quite fit in this file anymore.

Any place where use used to call `postCanEdit()` on the server, I just replaced with the equivalent of `userOwns(currentUser, post) || userCanDo(currentUser, 'posts.edit.all') || userIsPostGroupOrganizer(currentUser, post)`.

`postCanDelete()` was only used in one place.

Reply:
Seems fine. Some day this may get refactored, and on that day searching for `posts.edit.all` will find all the places.